extensions/amp-story-captions/0.1/amp-story-captions.js

```diff
@@ -27,8 +36,22 @@ export class AmpStoryCaptions extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.element.appendChild(this.container_);
+    this.container_ = <div />;
+    this.checkAndMaybeSetPreset_();
+  }
+  /**
+   * Checks if data preset is defined and valid.
+   * If valid it renders in shadow dom with preset syles.
+   * @private
+   */
+  checkAndMaybeSetPreset_() {
+    const preset = this.element.getAttribute('data-preset');
+    if (presetValues.includes(preset)) {
+      this.container_.classList.add(`amp-story-captions-${preset}`);
+      createShadowRootWithStyle(this.element, this.container_, PRESETS_CSS);
```

Comment:
Are there any CSS rules that we wouldn't want the captions component to inherit when there are presets (because they could break the styles)? Otherwise, we could let people override any part of the presets independently (not use shadow dom) without having to use CSS vars to support specific modifications to the presets (effectively using the presets as starting points for their styles).

It could be a really interesting mechanic to provide the presets as base styles that others could change some parts only (the background, color, font family, font weight, text transform, etc)

Reply:
The presets would mostly break without the shadow DOM boundaries. They'd either break or look different based on publisher provided CSS, which is not acceptable.

Rendering presets in shadow DOM, and no preset (custom) in light DOM SGTM.